js/index.js

```diff
@@ -33,51 +33,154 @@ function renderEverything() {
+  } else {
+    sauce.style.visibility = "hidden";
+  }
 }
 
 function renderGlutenFreeCrust() {
-  // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  const gluten = document.querySelector(".crust");
+  if (state.glutenFreeCrust) {
+    gluten.classList.remove("crust-gluten-free");
+  } else {
+    gluten.classList.add("crust-gluten-free");
+  }
 }
 
 function renderButtons() {
-  // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  const btnPepperoni = document.querySelector(".btn.btn-pepperoni");
+  if (!state.pepperoni) {
+    btnPepperoni.classList.remove("active");
+  } else {
+    btnPepperoni.classList.add("active");
+  }
+  const btnMushrooms = document.querySelector(".btn.btn-mushrooms");
+  if (!state.mushrooms) {
+    btnMushrooms.classList.remove("active");
+  } else {
+    btnMushrooms.classList.add("active");
+  }
+  btnGreenPeppers = document.querySelector(".btn.btn-green-peppers");
+  if (!state.greenPeppers) {
+    btnGreenPeppers.classList.remove("active");
+  } else {
+    btnGreenPeppers.classList.add("active");
+  }
+  btnWhiteSauce = document.querySelector(".btn.btn-sauce");
+  if (!state.whiteSauce) {
+    btnWhiteSauce.classList.remove("active");
+  } else {
+    btnWhiteSauce.classList.add("active");
+  }
+  btnGlutenFree = document.querySelector(".btn.btn-crust");
+  if (!state.glutenFreeCrust) {
+    btnGlutenFree.classList.remove("active");
+  } else {
+    btnGlutenFree.classList.add("active");
+  }
 }
 
 function renderPrice() {
-  // Iteration 4: change the HTML of `<aside class="panel price">`
+  const ingredientsP = document.querySelector("strong");
+  const pepperoni = document.querySelector("#pepperoni-price");
+  const mushrooms = document.querySelector("#mushrooms-price");
+  const peppers = document.querySelector("#peppers-price");
+  const sauce = document.querySelector("#sauce-price");
+  const gluten = document.querySelector("#gluten-price");
+  const pepperoniP = ingredients.pepperoni.price;
+  const mushroomsP = ingredients.mushrooms.price;
+  const peppersP = ingredients.greenPeppers.price;
+  const sauceP = ingredients.whiteSauce.price;
+  const glutenP = ingredients.glutenFreeCrust.price;
+  if (state.pepperoni) {
+    pepperoni.style.display = "block";
+    basePrice += pepperoniP;
+  } else {
+    pepperoni.style.display = "none";
+  }
+  if (state.mushrooms) {
+    mushrooms.style.display = "block";
+    basePrice += mushroomsP;
+  } else {
+    mushrooms.style.display = "none";
+  }
+  if (state.greenPeppers) {
+    peppers.style.display = "block";
+    basePrice += peppersP;
+  } else {
+    peppers.style.display = "none";
+  }
+  if (state.whiteSauce) {
+    sauce.style.display = "block";
+    basePrice += sauceP;
+  } else {
+    sauce.style.display = "none";
+  }
+  if (state.glutenFreeCrust) {
+    gluten.style.display = "block";
+    basePrice += glutenP;
+  } else {
+    gluten.style.display = "none";
+  }
```

Comment:
Muy bien! Aquí también puedes evitar redundancias, usando un bucle for... in de la siguiente forma:
```
for(let oneIngredient in ingredients) {
    if(state[oneIngredient]) {
       // ...
     } else {
      // ...
    }
}
```